packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js

```diff
@@ -1,23 +1,52 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import ViewDropdown from '../../components/ViewDropDown/ViewDropdown.component';
+import { apiURL } from '../../apiURL';
+import PropTypes from 'prop-types';
 import './DropdownGroup.Style.css';
 
-export const DropdownGroup = () => {
-  const Colors = ['Black', 'White', 'Blue', 'Half-white'];
-  const Sizes = ['Small', 'Medium', 'Large', 'XL', 'XXL'];
-  const Quantities = [1, 2, 3, 4, 5, 6, 7, 8, 9];
+export const DropdownGroup = ({ productId }) => {
+  const [variants, setVariants] = useState([]);
+  const colors = [];
+  const sizes = [];
+  const quantities = [];
+
+  useEffect(() => {
+    const fetchVariantsByProductId = async () => {
+      const response = await fetch(`${apiURL()}/variants?product=${productId}`);
+      const variantsProductData = await response.json();
+      setVariants(variantsProductData);
+    };
+
+    fetchVariantsByProductId();
+  }, [productId, variants]);
+
+  variants.forEach((variant) => {
+    if (!colors.includes(variant.color)) {
+      colors.push(variant.color);
+    }
+    if (!sizes.includes(variant.size)) {
+      sizes.push(variant.size);
+    }
+    if (!quantities.includes(variant.stock)) {
+      quantities.push(variant.stock);
+    }
```

Comment:
I would go back to Milton's suggestion here to use Set. 
There were 2 thumbs up on that comment, but no reply and no change either. Sneaky! 😅 


<img width="722" alt="image" src="https://user-images.githubusercontent.com/64137994/207686609-21957136-9d21-4fff-ad6a-d83b57d6c7bc.png">

Reply:
I thought milton commented as I didn't use usestate for variants , after fetching I have used usestate for variants. sorry, will change now